app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -6,12 +6,12 @@ def available_sso
   end
 
   def authenticate
-    return true if (User.current && Rails.env.test? && api_request?) ||
-                   session[:user] && (User.current = User.unscoped.find_by(id: session[:user]))
+    return true if User.current && Rails.env.test? && api_request?
+    return !User.current.disabled? if session[:user] && (User.current = User.unscoped.find_by(id: session[:user]))
```

Comment:
Thanks @rahulbajaj0509 for the notes! :) I do not agree with this, the User.current.disabled is an attribute of the `ActiveRecord`, thus very fast if it's already loaded in memory, what it is in here. There is no point in caching in, because it'd cache on disk, what would be even slower.

Reply:
I think the suggestion was to use memoize patter `||=`, not using Rails.cache (that may end up written on disk or stored in redis). However I agree, I don't see a big value in memoizing this. Rahul is right that each request runs this method, but at the same time, memoization holds just for single reqest. We wouldn't get much out of it.

Reply:
Exactly, memoization by `||=` is useless across requests, so I've directly assumed the Rails cache :) but there's not much value in memory memoization eighter.

Reply:
I was thinking more in the terms of using the `defined?` keyword (check link in the comment above) but as I can see it is not much of a use here, i guess.

Reply:
I don't think that's is great example to use the memoize. As @ares said it holds for single request. The solution for that is write something like a cache but it will be better to use a Rails cache then. So I my opinion is not to use memoize in this case. Any opinions on that?

Reply:
I think we agreed there is no need for that.